common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -423,6 +425,20 @@ def _linked_dir_for(path)
           max_by(&:length)
       end
 
+      def _clone_repo_contents(target_directory:)
+        SharedHelpers.with_git_configured(credentials: credentials) do
+          path = target_directory || File.join("tmp", source.repo)
+          return path if Dir.exist?(File.join(path, ".git"))
```

Comment:
Hmm, yeah I see your concern, we could introduce some cleanup after Dependabot is completed? Not sure where that would fit in, but can investigate

Reply:
I was thinking something like
```
if Dir.exist?(File.join(path, ".git"))
   SharedHelpers.run_shell_command("git -C #{path} checkout #{source.branch}")
end
```

But due to the `--depth=1`, we won't have that ref in this clone 😿 . We'd need to `git fetch && git checkout` to ensure the existing repo's on the requested branch? (maybe that's why actions does an init&&fetch&&checkout instead of clone? 🤷 ).

Per top-level comment, this is good enough for v0: maybe just an `// XXX: assume we're retrying the branch so the repo is in the same state` or some other from of "here be dragons" comment?